contracts/p1/mixins/RecollateralizationLib.sol

```diff
@@ -158,31 +168,37 @@ library RecollateralizationLibP1 {
                 bal = bal.plus(reg.assets[i].bal(address(ctx.stRSR)));
             }
 
-            // Ignore dust amounts for assets not in the basket; their value is inaccessible
-            // {tok} = {tok/BU} * {BU}
-            uint192 inBasket = ctx.bh.quantity(reg.erc20s[i]).mul(ctx.basketsHeld, FLOOR);
-            if (bal < inBasket) inBasket = bal; // not sure if needed
+            uint192 q = ctx.bh.quantity(reg.erc20s[i]); // {tok/BU}
 
             // Skip over dust-balance assets not in the basket
             {
                 (uint192 lotLow, ) = reg.assets[i].lotPrice(); // {UoA/tok}
 
                 // Intentionally include value of IFFY/DISABLED collateral
                 if (
-                    inBasket == 0 &&
+                    q == 0 &&
                     !TradeLib.isEnoughToSell(reg.assets[i], bal, lotLow, ctx.minTradeVolume)
                 ) continue;
             }
 
             (uint192 low, uint192 high) = reg.assets[i].price(); // {UoA/tok}
 
-            assert(high != FIX_MAX || inBasket == 0); // collateral in the basket must be priced
+            assert(high != FIX_MAX || q == 0); // collateral in the basket must be priced
 
             // throughout this section +/- is same as Fix.plus/Fix.minus
 
+            // Ignore dust amounts for assets not in the basket; their value is inaccessible
+            // {tok} = {tok/BU} * {BU}
+            uint192 inBasket = q.mul(ctx.basketsHeld, FLOOR);
+            if (bal < inBasket) inBasket = bal; // not sure if needed, might be unreachable
```

Comment:
i feel like if we execute this if result then something has gone horribly wrong, no? can we really ever have a balance less than whats in the current basket?

Reply:
Agreed, shouldn't happen. 

`BasketHandler.basketsHeldBy` computes `basketsHeld` defensively: it does a CEIL in computation of the quantity: `uint192 q = basket.refAmts[basket.erc20s[i]].div(refPerTok, CEIL);` and a FLOOR in computation of baskets: `baskets = fixMin(baskets, bal.div(q));`(div without specification performs a FLOOR).

So if we take this `basketsHeld` and multiply by q, and take the FLOOR, this should be <= the token balance. Ok, agreed. I'm going to delete this in P1 and in P0 put an assert, and maybe fuzz finds a way to find it.

Reply:
yeah i think an assert is the way to go here